librad/src/keys.rs

```diff
@@ -339,35 +333,35 @@ impl<'b> minicbor::Decode<'b> for PublicKey {
         }
 
         let data = d.bytes()?;
-        ed25519::PublicKey::from_slice(data)
+        ed25519::VerificationKeyBytes::try_from(data)
             .map(PublicKey)
-            .ok_or(minicbor::decode::Error::Message(
-                "Invalid length for ed25519 public key",
-            ))
+            .map_err(|_| minicbor::decode::Error::Message("Invalid length for ed25519 public key"))
     }
 }
 
 // Signature
 
 impl Signature {
     pub fn verify(&self, data: &[u8], pk: &PublicKey) -> bool {
-        ed25519::verify_detached(self, &data, pk)
+        pk.verify(self, data)
     }
 
     pub fn from_bs58(s: &str) -> Option<Self> {
-        ensure_initialised();
         let bytes = match bs58::decode(s.as_bytes())
             .with_alphabet(bs58::alphabet::BITCOIN)
             .into_vec()
         {
             Ok(v) => v,
             Err(_) => return None,
         };
-        sodiumoxide::crypto::sign::Signature::from_slice(&bytes).map(Self)
+        ed25519::Signature::try_from(bytes.as_slice())
+            .map(Self)
+            .ok()
     }
 
     pub fn to_bs58(&self) -> String {
-        bs58::encode(self.0)
+        let bytes: [u8; 64] = self.0.into();
+        bs58::encode(&bytes[..])
             .with_alphabet(bs58::alphabet::BITCOIN)
             .into_string()
     }
```

Comment:
This is ugly because zebra's `Signature` doesn't have an `as_ref() -> &[u8]` due to the inner data being split in two. There's an issue for it upstream.

Reply:
We could leave an inline TODO with a link to the upstream issue.

Reply:
This method can be considered deprecated, so nothing to worry about.